src/main/webapp/main/Main.js

```diff
@@ -25,10 +26,10 @@ class Main extends Component {
         projectReviewApproved: '',
         attachmentsApproved: ''
       },
-      consentKey: this.props.location.state !== undefined && this.props.location.state.consentKey !== undefined  ? this.props.location.state.consentKey : component.consentKey,
-      projectKey: this.props.location.state !== undefined && this.props.location.state.projectKey !== undefined  ? this.props.location.state.projectKey : component.projectKey,
-      tab: this.props.location.state !== undefined && this.props.location.state.tab !== undefined  ? this.props.location.state.tab : component.tab,
-      issueType: this.props.location.state !== undefined && this.props.location.state.issueType !== undefined  ? this.props.location.state.issueType : component.issueType
+      consentKey: params.get('consentKey') != null ? params.get('consentKey') : component.projectKey,
+      projectKey: params.get('projectKey') != null ? params.get('projectKey') : component.projectKey,
+      tab:  params.get('tab') != null ? params.get('tab') : component.tab,
```

Comment:
https://lodash.com/docs/4.17.15#defaultTo makes these kinds of ternaries a lot more readable. 
```
tab:  _.defaultTo(params.get('tab'), component.tab),
```

Reply:
Done!